src/core/ui.js

```diff
@@ -79,6 +81,19 @@ document.documentElement.addEventListener("click", () => {
 });
 respecUI.appendChild(menu);
 
+menu.addEventListener("keydown", e => {
+  if (e.key === "Escape" && !menu.hidden) {
+    menu.classList.add("respec-hidden");
+    menu.classList.remove("respec-visible");
+    respecPill.setAttribute("aria-expanded", String(menu.hidden));
+    respecPill.focus();
+    menu.hidden = true;
+  }
+});
+window.addEventListener("load", () => {
```

Comment:
Consider, we only want to trap focus when the menu is visible, otherwise we need to release focus. See where the click even to open the menu happens.

Reply:
In that case, the menu.removeEventListener has also to be written?